### PR TITLE
app platform: app rollback api spec

### PR DESCRIPTION
**Pull request by kamaln7**
This adds API documentation for the App Rollbacks feature.

~Ready for review but please do not merge yet.~ :shipit:

### CODE REVIEW

**Review by ChiefMateStarbuck — COMMENTED** [review]
Looks good! Thank you! An approval from another APP PLATFORM team member is not necessary but would be helpful.

**Comment by kamaln7** [comment]
Thanks @ChiefMateStarbuck! I'm not sure how to address these two linter errors, any ideas?

`commit_app_rollback`
```
specification/resources/apps/commit_app_rollback.yml
 1:1  error  operationid-must-follow-naming-conventions  POST /v2/apps/{app_id}/rollback/commit - commit_app_rollback should start with one of: create, post, add, tag, install, reset, upgrade, recycle, run, retry, validate, assign. Prefer 'create' for new resources. Custom verbs may be acceptable for clarity.
```

`revert_app_rollback`
```
specification/resources/apps/revert_app_rollback.yml
 1:1  error  operationid-must-follow-naming-conventions  POST /v2/apps/{app_id}/rollback/revert - revert_app_rollback should start with one of: create, post, add, tag, install, reset, upgrade, recycle, run, retry, validate, assign. Prefer 'create' for new resources. Custom verbs may be acceptable for clarity.
```

File: specification/resources/apps/models/app.yml [hunk 1]
```diff
@@ -67,6 +67,10 @@ properties:
     title: Time of the app's last configuration update
     type: string
     example: 2020-12-01T00:42:16Z
+  pinned_deployment:
+    allOf:
+      - description: The deployment that the app is pinned to
```
Comment (nicktate) [inline specification/resources/apps/models/app.yml:72]: nit: -- I tried to look around and it seems like we end with a period for other `description` fields
```suggestion
      - description: The deployment that the app is pinned to.
```

**Review by nicktate — APPROVED** [review]

**Review by karanthukral — DISMISSED** [review]
LGTM

**Comment by ChiefMateStarbuck** [comment]
> Thanks @ChiefMateStarbuck! I'm not sure how to address these two linter errors, any ideas?
> 
> `commit_app_rollback`
> 
> ```
> specification/resources/apps/commit_app_rollback.yml
>  1:1  error  operationid-must-follow-naming-conventions  POST /v2/apps/{app_id}/rollback/commit - commit_app_rollback should start with one of: create, post, add, tag, install, reset, upgrade, recycle, run, retry, validate, assign. Prefer 'create' for new resources. Custom verbs may be acceptable for clarity.
> ```
> 
> `revert_app_rollback`
> 
> ```
> specification/resources/apps/revert_app_rollback.yml
>  1:1  error  operationid-must-follow-naming-conventions  POST /v2/apps/{app_id}/rollback/revert - revert_app_rollback should start with one of: create, post, add, tag, install, reset, upgrade, recycle, run, retry, validate, assign. Prefer 'create' for new resources. Custom verbs may be acceptable for clarity.
> ```
The `operationID` for any endpoint must start with one of the following `create, post, add, tag, install, reset, upgrade, recycle, run, retry, validate, assign`.

Currently, `revert_app_rollback` and `commi_app_rollback` do not fit this criteria (`revert` and `commit` are not key words.)

Ill post an example of the change. It would also be clean to change the names of the files as well, including the examples.

File: specification/resources/apps/commit_app_rollback.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+operationId: commit_app_rollback
```
Comment (ChiefMateStarbuck) [inline specification/resources/apps/commit_app_rollback.yml:1]:
```suggestion
operationId: run_commit_app_rollback
```

`run` is a keyword that can prefix any operationID

File: specification/resources/apps/revert_app_rollback.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
+operationId: revert_app_rollback
```
Comment (ChiefMateStarbuck) [inline specification/resources/apps/revert_app_rollback.yml:1]:
```suggestion
operationId: run_revert_app_rollback
```

**Comment by kamaln7** [comment]
Thanks @ChiefMateStarbuck

**Comment by kamaln7** [comment]
Hey @ChiefMateStarbuck this is ready for review again. Thanks!

**Review by ChiefMateStarbuck — APPROVED** [review]

**Pull request merged**